actors/sunbird-lms-mw/actors/user/src/main/java/org/sunbird/user/actors/UserManagementActor.java

```diff
@@ -287,17 +287,30 @@ private void updateUser(Request actorMessage) {
    *
    * @param userMap
    */
-  private void encryptExternalDetails(Map<String, Object> userMap) {
+  private void encryptExternalDetails(Map<String, Object> userMap,Map<String,Object> userDbRecords) {
     List<Map<String, Object>> extList =
         (List<Map<String, Object>>) userMap.get(JsonKey.EXTERNAL_IDS);
     if (!(extList == null || extList.isEmpty())) {
       extList.forEach(
           map -> {
             try {
-              if (map.get(JsonKey.ID_TYPE).equals(JsonKey.DECLARED_EMAIL)
-                  || map.get(JsonKey.ID_TYPE).equals(JsonKey.DECLARED_PHONE)) {
-                map.put(JsonKey.ID, UserUtility.encryptData((String) map.get(JsonKey.ID)));
+              String idType = (String)map.get(JsonKey.ID_TYPE);
+              switch (idType){
+                case JsonKey.DECLARED_EMAIL:
+                case JsonKey.DECLARED_PHONE:
+                         if(UserUtility.isMasked((String)map.get(JsonKey.ID))){
```

Comment:
Can this line be lifted up before switch case?
Not sure looks left code is much readable and we need have a simple if within it, I'd imagine.

`if (map.get(JsonKey.ID_TYPE).equals(JsonKey.DECLARED_EMAIL)	             
                  || map.get(JsonKey.ID_TYPE).equals(JsonKey.DECLARED_PHONE)) {
      if (isMasked) {
          // first time and user hasn't changed any
          // pick value from existing email/phone column
      } else {
      }
}`

Reply:
Added comments as per review comment